store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/utils/ExtractKeyFromRow.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2017. Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.parquetstore.utils;
+
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.sql.Row;
+import scala.collection.Seq;
+import scala.collection.Seq$;
+import scala.collection.mutable.Builder;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ *
+ */
+public class ExtractKeyFromRow implements Function<Row, Seq<Object>>, Serializable {
+
+    private static final long serialVersionUID = -5811180619204002981L;
+    private final HashSet<String> groupByColumns;
+
+    public ExtractKeyFromRow(final Set<String> groupByColumns, final HashMap<String, String[]> columnToPaths, final boolean isEntity, final HashMap<String, String> propertyToAggregatorMap) {
```

Comment:
Please can you change the parameter types to be Map instead HashMap and also check for this issue in your other changes.